test/integration/response-test.coffee

```diff
@@ -0,0 +1,341 @@
+      )
+
+      it('evaluates the responses as valid', ->
+        assert.deepInclude(runtimeInfo.dredd.stats, {tests: 2, passes: 2})
+      )
+    )
+
+    describe('when the server returns empty responses', ->
+      runtimeInfo = undefined
+
+      beforeEach((done) ->
+        app = createServer()
+        app.get('/resource.json', (req, res) ->
+          res.type('json').send()
+        )
+        app.get('/resource.csv', (req, res) ->
+          res.type('text/csv').send()
+        )
+        dredd = new Dredd({options: {path: apiDescription.path}})
+        runDreddWithServer(dredd, app, (err, info) ->
+          runtimeInfo = info
+          done(err)
+        )
+      )
+
+      it('evaluates the responses as valid', ->
+        assert.deepInclude(runtimeInfo.dredd.stats, {tests: 2, passes: 2})
+      )
+    )
+  )
+)
+
+
+[
+    name: 'API Blueprint'
+    path: './test/fixtures/response/empty-body.apib'
+  ,
+    name: 'Swagger'
+    path: './test/fixtures/response/empty-body.yaml'
+].forEach((apiDescription) ->
+  describe("Specifying no response body in the #{apiDescription.name}, but specifying a schema", ->
+    describe('when the server returns a response not valid according to the schema', ->
+      runtimeInfo = undefined
+
+      beforeEach((done) ->
+        app = createServer()
+        app.get('/resource', (req, res) ->
+          res.json({name: 123})
+        )
+        dredd = new Dredd({options: {path: apiDescription.path}})
+        runDreddWithServer(dredd, app, (err, info) ->
+          runtimeInfo = info
+          done(err)
+        )
+      )
+
+      it('evaluates the response as invalid', ->
+        assert.deepInclude(runtimeInfo.dredd.stats, {tests: 1, failures: 1})
+      )
+      it('prints JSON Schema validation error', ->
+        assert.include(runtimeInfo.dredd.logging, 'At \'/name\' Invalid type: number (expected string)')
+      )
+    )
+
+    describe('when the server returns a response valid according to the schema', ->
+      runtimeInfo = undefined
+
+      beforeEach((done) ->
+        app = createServer()
+        app.get('/resource', (req, res) ->
+          res.json({name: "test"})
+        )
+        dredd = new Dredd({options: {path: apiDescription.path}})
+        runDreddWithServer(dredd, app, (err, info) ->
+          runtimeInfo = info
+          done(err)
+        )
+      )
+
+      it('evaluates the response as valid', ->
+        assert.deepInclude(runtimeInfo.dredd.stats, {tests: 1, passes: 1})
+      )
+    )
+  )
+)
+
+
+[
+    name: 'API Blueprint'
+    path: './test/fixtures/response/empty-body-empty-schema.apib'
+  ,
+    name: 'Swagger'
+    path: './test/fixtures/response/empty-body-empty-schema.yaml'
+].forEach((apiDescription) ->
+  describe("Specifying no response body in the #{apiDescription.name} and having hooks ensuring empty response", ->
+    describe('when the server returns a non-empty responses', ->
+      runtimeInfo = undefined
+
+      beforeEach((done) ->
```

Comment:
Same as previous comment (move server creation to upper scope and reuse it for all assertions, since it is the same for all of them).